File: HackerRank/DataStructures/LinkedLists/InsertAtTail.java
static SinglyLinkedListNode insertNodeAtTail(SinglyLinkedListNode head, int data) {
    SinglyLinkedListNode newnode = new SinglyLinkedListNode(data);
    
    if(head == null){
        head = newnode;
        return head;
    }
    
    SinglyLinkedListNode temp = head;
    while(temp.next != null){
        temp = temp.next;
    }
    temp.next = newnode;
    return head;
}

File: HackerRank/DataStructures/LinkedLists/insert_at_tail.cpp
SinglyLinkedListNode* insertNodeAtTail(SinglyLinkedListNode* head, int data) {
    SinglyLinkedListNode* newnode = new SinglyLinkedListNode(data);
    
    if(head == nullptr){
        head = newnode;
        return head;
    }
    
    SinglyLinkedListNode* temp = head;
    while(temp -> next != nullptr){
        temp = temp -> next;
    }
    temp -> next = newnode;
    return head;
}